code/game/objects/items/melee/baton.dm

```diff
@@ -0,0 +1,656 @@
+	/// How much stamina damage we deal on a successful hit against a living, non-cyborg mob.
+	var/stamina_damage = 55
+	/// Can we stun cyborgs?
+	var/affect_cyborg = FALSE
+	/// The path of the default sound to play when we stun something.
+	var/on_stun_sound = 'sound/effects/woodhit.ogg'
+	/// The volume of the above.
+	var/on_stun_volume = 75
+	/// Do we animate the "hit" when stunning something?
+	var/stun_animation = TRUE
+	/// Whether the stun attack is logged. Only relevant for abductor batons, which have different modes.
+	var/log_stun_attack = TRUE
+
+/obj/item/melee/baton/Initialize()
+	. = ..()
+	// Adding an extra break for the sake of presentation
+	if(stamina_damage != 0)
+		offensive_notes = "\nVarious interviewed security forces report being able to beat criminals into exhaustion with only [span_warning("[round(100 / stamina_damage, 0.1)] hit\s!")]"
+
+/**
+ * Ok, think of baton attacks like a melee attack chain:
+ *
+ * [/baton_attack()] comes first. It checks if the user is clumsy, if the target parried the attack and handles some messages and sounds.
+ * * Depending on its return value, it'll either do a normal attack, continue to the next step or stop the attack.
+ *
+ * [/finalize_baton_attack()] is then called. It handles logging stuff, sound effects and calls baton_effect().
+ * * The proc is also called in other situations such as stunbatons right clicking or throw impact. Basically when baton_attack()
+ * * checks are either redundant or unnecessary.
+ *
+ * [/baton_effect()] is third in the line. It knockdowns targets, along other effects called in additional_effects_cyborg() and
+ * * additional_effects_non_cyborg().
+ *
+ * Last but not least [/set_batoned()], which gives the target the IWASBATONED trait with REF(user) as source and then removes it
+ * * after a cooldown has passed. Basically, it stops users from cheesing the cooldowns by dual wielding batons.
+ *
+ * TL;DR: [/baton_attack()] -> [/finalize_baton_attack()] -> [/baton_effect()] -> [/set_batoned()]
+ */
+/obj/item/melee/baton/attack(mob/living/target, mob/living/user, params)
+	add_fingerprint(user)
+	var/list/modifiers = params2list(params)
+	switch(baton_attack(target, user, modifiers))
+		if(BATON_DO_NORMAL_ATTACK)
+			return ..()
+		if(BATON_ATTACKING)
+			finalize_baton_attack(target, user, modifiers)
+
+/obj/item/melee/baton/proc/baton_attack(mob/living/target, mob/living/user, modifiers)
+	. = BATON_ATTACKING
+
+	if(clumsy_check(user, target))
+		return BATON_ATTACK_DONE
+
+	if(!active || LAZYACCESS(modifiers, RIGHT_CLICK))
+		return BATON_DO_NORMAL_ATTACK
+
+	if(cooldown_check > world.time)
+		var/wait_desc = get_wait_description()
+		if (wait_desc)
+			to_chat(user, wait_desc)
+		return BATON_ATTACK_DONE
+
+	if(check_parried(target, user))
+		return BATON_ATTACK_DONE
+
+	if(HAS_TRAIT_FROM(target, TRAIT_IWASBATONED, REF(user))) //no doublebaton abuse anon!
+		to_chat(user, span_danger("You fumble and miss [target]!"))
+		return BATON_ATTACK_DONE
```

Comment:
It doesn't look like it enters into a cooldown here at all if the baton fails to hit someone.

Reply:
It's been like this since before this refactor.

Reply:
okay